### ISSUE TITLE
Metadata on expenses

### ISSUE THREAD

**Issue by aerugo** [opencollective/opencollective#6038]
TL;DR
**Allow storing metadata on expenses that are intended to be read and edited by other third party services, and allow querying expenses through the API filtered by this metadata.**

We are planning an integration of expense management between Cobudget and Open Collective. This integration will be almost completely on the Cobudget side, and will enable users of Cobudget to upload their expenses directly through the Cobudget platform if the admin has connected the round on Cobudget to a collective. We also want to show the expenses related to a bucket in a list on the bucket page and summarize the total amounts that have already been submitted and/or paid out.

To do this, we need to be able to connect a bucket to a set of expenses on OC and to query expenses related to a certain bucket.

Cobudget has three "levels" in its hierarchy: 
Groups, rounds, and buckets.

For a collective with lots of activities, a round probably maps most closely to a project. For example, Blivande has a [monthly budget available to buy tools for its makerspace](https://dreams.blivande.com/c/tau-tools-and-upgrades), but also has another budget available for [container refurbishment](https://dreams.blivande.com/c/frihamnstorget-modules). The Borderland has a [new budget for dream projects](https://dreams.theborderland.se/c/borderland-dreams-2022) every year with slightly varying questions and criteria, and also has a separate budget available for infrastructure.

To use Open Collective for the Borderland without an integration, we solved this by using tags on Open Collective in an unintended way. We asked users to tag their expenses with the corresponding unique bucket 'slug' (example [bucket](https://opencollective.com/borderland/projects/the-borderland-2022/expenses?searchTerm=cl3n9cf2h402809l2714sc216) and corresponding [expenses on OC](https://github.com/opencollective/opencollective/issues/5458)). This works, but it also makes it impossible to use tags for their intended purpose - to classify expenses for visualization - as the IDs are not human-readable and there are over 150 of them.

My proposed solution is a simple hack: 
Allow storing metadata on expenses that are intended to be read and edited by other third party services, and allow querying expenses through the API filtered by this metadata.
